File: client/generated_rest_api/java/src/main/java/io/swagger/client/api/DefaultApi.java
package io.swagger.client.api;

import com.sun.jersey.api.client.GenericType;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;
import io.swagger.client.Pair;

import io.swagger.client.model.RobotCommand;
import io.swagger.client.model.RobotStatus;
import io.swagger.client.model.RobotConfiguration;
import io.swagger.client.model.RemoveCommand;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-04-30T12:28:53.526+02:00")
public class DefaultApi {
  private ApiClient apiClient;

  public DefaultApi() {
    this(Configuration.getDefaultApiClient());
  }

  public DefaultApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * 
   * send an action to a robot. The server will answer after the simulation turn is completed, and it will return the token for the next simulation turn.
   * @param command  (required)
   * @return RobotStatus
   * @throws ApiException if fails to make API call
   */
  public RobotStatus robotActionPost(RobotCommand command) throws ApiException {
    Object localVarPostBody = command;
    
    // verify the required parameter 'command' is set
    if (command == null) {
      throw new ApiException(400, "Missing the required parameter 'command' when calling robotActionPost");
    }
    
    // create path and map variables
    String localVarPath = "/robot-action".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    GenericType<RobotStatus> localVarReturnType = new GenericType<RobotStatus>() {};
    return apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
    
  }
  
  /**
   * 
   * Create a new robot.
   * @param configuration  (required)
   * @return RobotStatus
   * @throws ApiException if fails to make API call
   */
  public RobotStatus robotCreatePost(RobotConfiguration configuration) throws ApiException {
    Object localVarPostBody = configuration;
    
    // verify the required parameter 'configuration' is set
    if (configuration == null) {
      throw new ApiException(400, "Missing the required parameter 'configuration' when calling robotCreatePost");
    }
    
    // create path and map variables
    String localVarPath = "/robot-create".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    GenericType<RobotStatus> localVarReturnType = new GenericType<RobotStatus>() {};
    return apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
    
  }
  
  /**
   * 
   * Remove a robot.
   * @param remove  (required)
   * @throws ApiException if fails to make API call
   */
  public void robotRemovePost(RemoveCommand remove) throws ApiException {
    Object localVarPostBody = remove;
    
    // verify the required parameter 'remove' is set
    if (remove == null) {
      throw new ApiException(400, "Missing the required parameter 'remove' when calling robotRemovePost");
    }
    
    // create path and map variables
    String localVarPath = "/robot-remove".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    

    

    

    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] {  };

    
    apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, null);
    
  }
  
}

File: client/generated_rest_api/javascript/src/api/DefaultApi.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient', '../model/RobotCommand', '../model/RobotStatus', '../model/RobotConfiguration', '../model/RemoveCommand'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/RobotCommand'), require('../model/RobotStatus'), require('../model/RobotConfiguration'), require('../model/RemoveCommand'));
  } else {
    // Browser globals (root is window)
    if (!root.NetRobotsApi) {
      root.NetRobotsApi = {};
    }
    root.NetRobotsApi.DefaultApi = factory(root.NetRobotsApi.ApiClient, root.NetRobotsApi.RobotCommand, root.NetRobotsApi.RobotStatus, root.NetRobotsApi.RobotConfiguration, root.NetRobotsApi.RemoveCommand);
  }
}(this, function(ApiClient, RobotCommand, RobotStatus, RobotConfiguration, RemoveCommand) {
  'use strict';

  /**
   * Default service.
   * @module api/DefaultApi
   * @version 2.0
   */

  /**
   * Constructs a new DefaultApi. 
   * @alias module:api/DefaultApi
   * @class
   * @param {module:ApiClient} apiClient Optional API client implementation to use, default to {@link module:ApiClient#instance}
   * if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the robotActionPost operation.
     * @callback module:api/DefaultApi~robotActionPostCallback
     * @param {String} error Error message, if any.
     * @param {module:model/RobotStatus} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * send an action to a robot. The server will answer after the simulation turn is completed, and it will return the token for the next simulation turn.
     * @param {module:model/RobotCommand} command 
     * @param {module:api/DefaultApi~robotActionPostCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/RobotStatus}
     */
    this.robotActionPost = function(command, callback) {
      var postBody = command;

      // verify the required parameter 'command' is set
      if (command == undefined || command == null) {
        throw "Missing the required parameter 'command' when calling robotActionPost";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = RobotStatus;

      return this.apiClient.callApi(
        '/robot-action', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the robotCreatePost operation.
     * @callback module:api/DefaultApi~robotCreatePostCallback
     * @param {String} error Error message, if any.
     * @param {module:model/RobotStatus} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Create a new robot.
     * @param {module:model/RobotConfiguration} configuration 
     * @param {module:api/DefaultApi~robotCreatePostCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/RobotStatus}
     */
    this.robotCreatePost = function(configuration, callback) {
      var postBody = configuration;

      // verify the required parameter 'configuration' is set
      if (configuration == undefined || configuration == null) {
        throw "Missing the required parameter 'configuration' when calling robotCreatePost";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = RobotStatus;

      return this.apiClient.callApi(
        '/robot-create', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the robotRemovePost operation.
     * @callback module:api/DefaultApi~robotRemovePostCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Remove a robot.
     * @param {module:model/RemoveCommand} remove 
     * @param {module:api/DefaultApi~robotRemovePostCallback} callback The callback function, accepting three arguments: error, data, response
     */
    this.robotRemovePost = function(remove, callback) {
      var postBody = remove;

      // verify the required parameter 'remove' is set
      if (remove == undefined || remove == null) {
        throw "Missing the required parameter 'remove' when calling robotRemovePost";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = null;

      return this.apiClient.callApi(
        '/robot-remove', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
